detect-common-errors.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+directories="src tools tests"
```

Comment:
@dipanjanS there is no 'tools' directory in the repo. Please remove this from the directories' list.

Reply:
Same as above (added in this commit)